### PR TITLE
ci: Basic example CI

### LINKED ISSUE #697: Bad any cast with the ActsExampleGeometryDD4hep
Using the `ActsExampleGeometryDD4hep` result in a boost::bad_any_cast. I also tried to test it with `ActsExampleGeometryGeneric` and also encountered a boost::bad_any_cast.

Example command : `./../build/bin/ActsExampleGeometryDD4hep -n1 -j1 --mat-output-file geometry-map  --dd4hep-input ../Examples/Detectors/DD4hepDetector/compact/OpenDataDetector/OpenDataDetector.xml`

### PR DESCRIPTION
**Pull request by HadrienG2**
This is a first batch of Acts examples in CI, starting with those that I've been running in my private build for a while.

This PR also switches to a patched CI image that fixes #697 .

### CODE REVIEW

**Review by paulgessinger — COMMENTED** [review]
I would argue we probably want to put most of the scripting you now have in the YAML into a common script.

Apart from this: `-DACTS_LOG_FAILURE_THRESHOLD=WARNING` on jobs where we don't actually execute anything is probably not needed. Doesn't hurt of course, on the other hand.

**Comment by HadrienG2** [comment]
> I would argue we probably want to put most of the scripting you now have in the YAML into a common script.

Thanks, that's exactly the idea which I was missing to ease local testing.

> Apart from this: -DACTS_LOG_FAILURE_THRESHOLD=WARNING on jobs where we don't actually execute anything is probably not needed. Doesn't hurt of course, on the other hand.

Ah, yes, I didn't notice that we don't actually run anything on the CUDA and SyCL jobs... will fix that in #688 , which this part of the diff belongs to.

**Comment by HadrienG2** [comment]
Now that #688 is in, this one is ready for review.

File: .github/workflows/builds.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ env:
 jobs:
   linux:
     runs-on: ubuntu-latest
-    container: ghcr.io/acts-project/${{ matrix.image }}:v9
```
Comment (paulgessinger) [inline .github/workflows/builds.yml:17]: Can you bump this for all jobs?

**Review by paulgessinger — COMMENTED** [review]
Almost happy with this.

File: .github/workflows/builds.yml (hunk 1 above)
Comment (HadrienG2) [reply .github/workflows/builds.yml:17]: Done !

**Review by paulgessinger — APPROVED** [review]

**Pull request merged**